src/main/java/com/metamx/http/client/Request.java

```diff
@@ -57,8 +75,103 @@ public ChannelBuffer getContent()
     return content;
   }
 
-  public HttpResponseHandler<Intermediate, Final> getHandler()
+  public Request copy() {
+    Request retVal = new Request(method, url);
+    retVal.headers.putAll(this.headers);
```

Comment:
This doesn't do a deep copy of the Objects in headers so they are still shared references. Can we change Object to String to fix that? Netty is mostly just gonna toString them anyway. (It seems to have exceptions for Date and Calendar)